pytorch_lightning/core/memory.py

```diff
@@ -213,6 +216,18 @@ def out_sizes(self) -> List:
     def param_nums(self) -> List[int]:
         return [layer.num_parameters for layer in self._layer_summary.values()]
 
+    @property
+    def total_parameters(self) -> int:
+        return sum(p.numel() for p in self._model.parameters())
+
+    @property
+    def trainable_parameters(self) -> int:
+        return sum(p.numel() for p in self._model.parameters() if p.requires_grad)
+
+    @property
+    def model_size(self) -> float:
```

Comment:
What happens if a model has weights in different precision mode ?

Wont' model_size should check the precision of each weights individually ?

Reply:
Self.precision_bytes is calculated using self._model.precision 
So if model precision changes, model size changes!
:)

Reply:
@kartik4949 did you check if `model.precision` is assigned to 16 before the summary is called or not in case precision is set to 16 in Trainer?

Reply:
also can you add a simple test for that too?

Reply:
@rohitgr7 sure

Reply:
@rohitgr7 it wont change until the .fit method is called :)

Reply:
btw you think we should add this test in test_memory?

Reply:
yes, you can parametrize that test with `precision=32, 16` and add `pytest.skipif(gpu_available)` decorator to make sure it runs on GPU only.

Reply:
I did exactly above with skip if not native amp available too because of fp16  
i used evalmodeltemplate with max epochs 1
should we use the above or improve class precalculatedmodel with training_step and configure @rohitgr7 ?

Reply:
@rohitgr7 just checked model precision is set before summary gets called, so we are good on this.